src/elements/content-explorer/ContentExplorer.js

```diff
@@ -383,6 +389,27 @@ class ContentExplorer extends Component<Props, State> {
         }
     };
 
+    /**
+     * Updates item in currentCollection with the given thumbnailUrl
+     *
+     * @param {BoxItem} item - item that should have thumbnailUrl field set
+     * @param {number} index - index of item in this.state.currentCollection
+     * @param {string} thumbnailUrl - thumbnailUrl to populate field with
+     * @return {void}
+     */
+    assignItemThumbnailUrl = (item: BoxItem, index: number, thumbnailUrl: string): void => {
+        const newCollection = { ...this.state.currentCollection };
+        newCollection.items = newCollection.items.map((newItem, newIndex) => {
```

Comment:
Could also do this in one line `(newIndex === index) ? {...cloneDeep(item), thumbnailUrl} : newItem`